src/aap_eda/conf/registry.py

```diff
@@ -0,0 +1,190 @@
+
+from aap_eda.core.models import Setting
+from aap_eda.core.utils.crypto.base import SecretValue
+
+ENCRYPTED_STRING = "$encrypted$"
+
+
+class InvalidKeyError(Exception):
+    ...
+
+
+class InvalidValueError(Exception):
+    ...
+
+
+@dataclass
+class RegistryData(object):
+    name: str
+    default: Any
+    type: type = str
+    read_only: bool = False
+    encrypted: bool = False
+
+
+_APPLICATION_SETTING_REGISTRIES = [
+    RegistryData(
+        name="INSIGHTS_TRACKING_STATE",
+        type=bool,
+        default=False,
+    ),
+    RegistryData(
+        name="AUTOMATION_ANALYTICS_URL",
+        read_only=True,
+        default=settings.AUTOMATION_ANALYTICS_URL,
+    ),
+    RegistryData(
+        name="REDHAT_USERNAME",
+        default="",
+    ),
+    RegistryData(
+        name="REDHAT_PASSWORD",
+        encrypted=True,
+        default="",
+    ),
+    RegistryData(
+        name="SUBSCRIPTIONS_USERNAME",
+        default="",
+    ),
+    RegistryData(
+        name="SUBSCRIPTIONS_PASSWORD",
+        encrypted=True,
+        default="",
+    ),
+    RegistryData(
+        name="INSIGHTS_CERT_PATH",
+        read_only=True,
+        default=settings.INSIGHTS_CERT_PATH,
+    ),
+    RegistryData(
+        name="AUTOMATION_ANALYTICS_LAST_GATHER",
+        default="",
+    ),
+    RegistryData(
+        name="AUTOMATION_ANALYTICS_LAST_ENTRIES",
+        type=dict,
+        default={},
+    ),
+    RegistryData(
+        name="AUTOMATION_ANALYTICS_GATHER_INTERVAL",
+        type=int,
+        default=14400,
+    ),
+]
+
+
+class SettingsRegistry(object):
+    def __init__(self):
+        self._registry = OrderedDict()
+        for registry_data in _APPLICATION_SETTING_REGISTRIES:
+            self.register(registry_data)
+
+    def register(self, registry_data: RegistryData) -> None:
+        if registry_data.name in self._registry:
+            raise ImproperlyConfigured(
+                f"Setting {registry_data.name} is already registered."
+            )
+        self._registry[registry_data.name] = registry_data
+        if registry_data.read_only:
+            update_method = Setting.objects.update_or_create
+        else:
+            update_method = Setting.objects.get_or_create
+        update_method(
+            key=registry_data.name, defaults={"value": registry_data.default}
+        )
+
+    def get_registered_settings(
+        self, skip_read_only: bool = False
+    ) -> list[str]:
+        setting_names = []
+
+        for setting, data in self._registry.items():
+            if data.read_only and skip_read_only:
+                continue
+            setting_names.append(setting)
+        return setting_names
+
+    def is_setting_encrypted(self, key: str) -> bool:
+        return self._registry[key].encrypted
+
+    def is_setting_read_only(self, key: str) -> bool:
+        return self._registry[key].read_only
+
+    def get_setting_type(self, setting: str) -> type:
+        return self._registry[setting].type
```

Comment:
Minor: this should use "key" for consistency with the other methods, right?
```suggestion
    def get_setting_type(self, key: str) -> type:
        return self._registry[key].type
```